### PR TITLE
[DOC] Correction for doc guide + TOC fix in File

### PR DESCRIPTION
**Pull request by BurdetteLamar**
The doc guide is corrected to omit reference to GFM tables, which RDoc does not support.

In linking to verbatim example, I discovered an TOC jump-level that caused most of the headings in File doc to be omitted from the doc's left-TOC for the class. Now corrected.

### CODE REVIEW

File: doc/contributing/documentation_guide.md [hunk 1]
```diff
@@ -216,14 +216,12 @@ may not render them properly.
 In particular, avoid building tables with HTML tags
 (<tt><table></tt>, etc.).
 
-Alternatives are:
-
-- The GFM (GitHub Flavored Markdown) table extension,
-  which is enabled by default. See
-  {GFM tables extension}[https://github.github.com/gfm/#tables-extension-].
```
Comment (peterzhu2118) [inline doc/contributing/documentation_guide.md:223]: RDoc doe support GFM tables, but only in markdown files. For example, [doc/contributing/glossary.md](https://github.com/ruby/ruby/blob/master/doc/contributing/glossary.md) is generated at: https://docs.ruby-lang.org/en/master/contributing/glossary_md.html

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:223]: Thanks.  Can the doc for a single class/module/method/etc. be markdown?  Or must the entire file be markdown?  (Asking for a friend.)

Comment (peterzhu2118) [reply doc/contributing/documentation_guide.md:223]: According to [the readme](https://github.com/ruby/rdoc/blob/master/README.rdoc#writing-documentation-):

> You can also set the comment format for a single file through the `:markup:` directive, but this is only recommended if you wish to switch markup formats.

So I think you can switch to markdown by using `:markup: markdown`.

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:223]: Modified.

File: doc/contributing/documentation_guide.md [hunk 2]
```diff
@@ -216,16 +216,21 @@ may not render them properly.
...
+  will not be honored:
+
+    - Example {source}[https://github.com/ruby/ruby/blob/34d802f32f00df1ac0220b62f72605827c16bad8/file.c#L6570-L6596].
+    - Corresponding {output}[rdoc-ref:File@Read-2FWrite+Mode].
+
+- (Markdown format only): A {Github Flavored Markdown (GFM) table}[https://github.github.com/gfm/#tables-extension-],
+  using special formatting for the text:
+
+    - Example {source}[https://github.com/ruby/ruby/blob/34d802f32f00df1ac0220b62f72605827c16bad8/doc/contributing/glossary.md?plain=1].
+    - Corresponding {output}[https://docs.ruby-lang.org/en/master/contributing/glossary_md.html].
```
Comment (olleolleolle) [inline doc/contributing/documentation_guide.md:233]: These examples are well-chosen, and the links to the plain-text source variant is also thoughtful. 👍

**Review by olleolleolle — APPROVED** [review]
Thanks, @BurdetteLamar!

File: doc/contributing/documentation_guide.md (hunk 2 above)
Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:233]: Credit to @peterzhu2118 for the GFM example.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**